src/main/java/listeners/MainEventListener.java

```diff
@@ -33,16 +33,24 @@
     private final HashMap<String, Server> servers = new HashMap<>();
 
     /**
-     * Check if the given Member has administrator permissions. Notifies the user if they do not have permission.
+     * Check if the given Member has administrator permissions.
      * @param member The Member to check
      * @return True if the Member has admin, false otherwise
      */
-    private static boolean checkAdmin(Member member, TextChannel channel) {
-        if (!member.hasPermission(Permission.ADMINISTRATOR)) {
-            channel.sendMessage(member.getAsMention() + " You must have administrator permission to run this command.").queue();
-            return false;
-        }
-        return true;
+    private static boolean checkAdmin(Member member) {
+        return member.hasPermission(Permission.ADMINISTRATOR);
+    }
+
+    /**
+     * Check if the given Member has the role associated with a topic.
+     * @param member The Member to check
+     * @param topic The Topic to check
+     * @return True if the Member has permission, false otherwise
+     */
+    private static boolean checkRole(Member member, Topic topic) {
+        return member.getRoles()
+                .stream()
+                .anyMatch(r -> r.getName().equals(Server.TOPIC_PREFIX + topic.getName()));
```

Comment:
I think the topic prefix should be a part of the name already, looking at the `createTopic` function in `Server.java` -- unless it's removed somewhere else? Otherwise this check tries to account for a topic prefix that already exists in `topic.getName()`

Reply:
We have to append the prefix here because the name is stored in the model as, for example, `Java`, while the version stored in the server's role list is `Topic | Java`. Therefore, since we're checking against the list of roles, we have to search for `Topic | <name>` instead of just `<name>`.

Reply:
`displayName` vs. `name` method, `displayName` returns w/ Topic, name does not; extract to sep issue as covered in meeting

Reply:
Covered in #15